flax/core/scope.py

```diff
@@ -320,9 +322,9 @@ def reserve(self, name: str):
       name: the name to reserve.
     """
     if not isinstance(name, str):
-      raise NameTypeError(name)
+      raise errors.ScopeNameTypeError(name)
```

Comment:
I think wrong types should not have a separate error. Otherwise you could have many of this type of errors. I would prefer the standard `TypeError`

Reply:
I agree this one is probably not useful.